xarray/core/computation.py

```diff
@@ -42,6 +42,14 @@
 _JOINS_WITHOUT_FILL_VALUES = frozenset({"inner", "exact"})
 
 
+def _first_of_type(args, kind):
+    """" Returns either first object of type 'kind' or None if not found. """
```

Comment:
I'm sure I'm missing something but can this return None? Or it'll raise?

Reply:
It should not return None, and will raise if the logic is broken somehow. At a higher level in the code, there is dispatching based on the type of argument (to `apply_variable_ufunc` etc.), so this should be well behaved

Reply:
could you update the docstring, then? Something like
```suggestion
    """ Return either first object of type 'kind' or raise if not found. """
```
also, there might be a better choice than `NotImplementedError`. `ValueError`, maybe?